### ISSUE TITLE
Problem with SMS spoofing and pyopenssl

### ISSUE THREAD

**Issue by joshgreensmith** [trustedsec/social-engineer-toolkit#270]
I am running the social engineering toolkit on mac OSX 10.11.05 and I've had this issue with SMS spoofing, saying it is cause by an out of date version of pyopenssl, but I have updated and reinstalled it and still get the same error message below, is it just a problem with the API or something to do with my OS? Thanks a lot.

[*] Okay! Moving on - SET needs some information from you in order to spoof the message.
set:sms> Enter your email address for the spoofmytextmessage.com account:XXXXXXXX@gmail.com
[*] Note that the password below will be masked and you will not see the output.
set:sms> Enter your password for the spoofmytextmessage.com account:
[*] The next section requires a country code, this is the code you would use to dial to the specific country, for example if I was sending a message to 555-555-5555 to the United States (or from) you would enter +1 below.
set:sms> Enter the country code for the number you are sending TO (for example U.S would be '+1')[+1]:+44
set:sms> Enter the country code for the number you are sending FROM (for example U.S. would be '+1')[+1]:+44
set:sms> Enter the number to send the SMS TO - be sure to include country code (example: +15551234567):+44XXXXXXXXX
set:sms> Enter the number you want to come FROM - be sure to include country code (example: +15551234567):+44XXXXXXXXX
set:sms> Enter the message you want to send via the text message:Test
[*] Grabbing initial ID from spoofmytextmessage.com...
[!] Something went wrong while attempting to locate the site. Will print the error message below in a second.
[!] If you are getting an SSL23_GET_SERVER_HELLO:tlsv1 alert internal error, this is likely caused by an out of date version of pyopenssl.
[!] Recommend apt-get remove python-openssl, and run pip install pyopenssl to resolve this issue.
[!] Printing the error: [SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:590)

**Comment by trustedsec** [comment]
There is an open issue with pyopenssl and OSX not supporting the latest crypto.  This is primarily due to OS X shipping with an older version of Python. Would recommend updating Python to the latest 2.x or 3.x.

**Comment by Jpatrick2zo** [comment]
I tried this on Linux using Python 2.7.6 and got he same issue with OpenSSL. I did what SET tells you to by removing it with apt and then reinstalling it again through pip, but the error was still there. I read there was some issues with that version of Python handling SSL requests, so I upgraded to 2.7.11 and the issue in SET was gone, however this only lead me to another issue which was actually with the 3rd party service spoofmytextmessage.com. Attempting to send a text messaged directly on their website without using the API still didn't go through. I've opened a support ticket with them and waiting to hear back.
